flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphRestartTest.java

```diff
@@ -733,6 +733,51 @@ public void testFailureWhileRestarting() throws Exception {
 		assertEquals(JobStatus.RUNNING, executionGraph.getState());
 	}
 
+	@Test
+	public void testFailingOfTheSameExecutionWhileRestarting() throws Exception {
```

Comment:
I think it would be nice if the test name states the unit under test, the state and the expected outcome. E.g. it could be named `failGlobalIfTaskIsStillRunning_failingAnExecutionTwice_ShouldTriggerOnlyOneFailover`